docs/fetch-design-system-pages.js

```diff
@@ -0,0 +1,89 @@
+ *
+ * It also checks if the link is for an image or not, because they need different handling
+ * The capture groups for the above link would be 1. '!' 2. 'link' and 3. '../images/image.png'
+ *
+ * Check it out here: https://regex101.com/r/y5o9Tf/6
+ *
+ */
+const mdRelativeLinkRegex = /(!)?\[(.*?)\]\((?!http|#)(.*?)\)/g
+const rawGithubUrl =
+    'https://raw.githubusercontent.com/dhis2/design-system/master/principles/'
+const regularGithubUrl =
+    'https://github.com/dhis2/design-system/tree/master/principles/'
+
+// eslint-disable-next-line max-params
+const linkReplacer = (_, imageBang, text, linkPath) => {
+    // Different links for images or markdown pages
+    const githubUrl = imageBang ? rawGithubUrl : regularGithubUrl
+    const prefix = imageBang || '' // '!' or ''
+    const url = new URL(linkPath, githubUrl).href
+    return `${prefix}[${text}](${url})`
+}
+
+const replaceRelativeLinksWithGithubLinks = text =>
+    text.replace(mdRelativeLinkRegex, linkReplacer)
+
+// Template for creating .stories.mdx file for storybook that imports from regular markdown file
+const mdxTemplate = basename => `import { Meta, Description } from '@storybook/addon-docs/blocks'
+import ${startCase(basename).replace(/ /g, '')} from './${basename}.md'
+
+<Meta title="Design System Principles/${startCase(basename)}" />
+
+<Description>{${startCase(basename).replace(/ /g, '')}}</Description>
+`
+
+const outputDir = path.join('docs', 'design-system')
+
+const downloadAndProcessFiles = async filteredFiles => {
+    // Create output dir if one doesn't exist yet
+    await fs.mkdir(outputDir, { recursive: true })
+    return Promise.all(
+        filteredFiles.map(async ({ name, download_url }) => {
+            // Get file contents
+            const text = await fetch(download_url).then(res => res.text())
+            // Transform relative links (see comments above)
+            const processedText = replaceRelativeLinksWithGithubLinks(text)
+            // Make a regular markdown file from contents
+            await fs.writeFile(path.join(outputDir, name), processedText)
+            // Make an MDX stories file that imports regular markdown
+            const basename = path.basename(name, '.md')
+            await fs.writeFile(
+                path.join(outputDir, `${basename}.stories.mdx`),
+                mdxTemplate(basename)
+            )
+        })
+    )
+}
+
+// Fetch design system 'principles' directory info from github
+fetch(
+    'https://api.github.com/repos/dhis2/design-system/contents/principles?ref=master'
+)
+    .then(res => res.json())
+    // Filter out non-markdown files
+    .then(fileInfo =>
+        fileInfo.filter(
+            ({ name, type }) => type === 'file' && name.endsWith('.md')
+        )
+    )
+    // Fetch files and create local ones for storybook
+    .then(downloadAndProcessFiles)
+    .then(() =>
+        console.log('Design system files downloaded & processed for storybook')
+    )
+    .catch(console.error)
```

Comment:
Catching the error and logging it to the console implies we only want to notify about this failure. I think for something like this, that does probably make sense: we can still showcase our components without including the design system files. However, this does imply that the storybook demo is still functional with these and I have not verified this.

Also, I think we should probably include some additional information about the context before printing this error to the console, i.e.:
> The following error occurred while downloading and processing design system files. The design system will not be included in the demo storybook.
(and then the error)

Reply:
That's a good point -- if there's something that could go wrong that would interfere with the rest of the storybook, it would be partially-made story files that could cause run-time errors. In that case, if there's an error, maybe the directory the design system files get copied into should get wiped to delete any malformed files